pkg/reconciler/taskrun/resources/apply.go

```diff
@@ -157,8 +157,8 @@ func ApplyWorkspaces(ctx context.Context, spec *v1.TaskSpec, declarations []v1.W
 			stringReplacements[prefix+"path"] = ""
 		} else {
 			stringReplacements[prefix+"bound"] = "true"
-			alphaAPIEnabled := config.FromContextOrDefaults(ctx).FeatureFlags.EnableAPIFields == config.AlphaAPIFields
-			if alphaAPIEnabled {
+			betaAPIEnabled := config.FromContextOrDefaults(ctx).FeatureFlags.EnableAPIFields == config.BetaAPIFields
```

Comment:
it looks like this will stop working for users who have alpha fields enabled, no?

I'm also not sure it makes sense to gate implementation (rather than just validation) on alpha/beta/stable. If we've accepted a taskrun w/ isolated workspaces, I don't think it makes sense to silently fail to execute it if enable-api-fields is set to a bad value.

Reply:
Both good points... it's strange that is was implemented like this, we usually only gate features at validation.